Cargo.toml

```diff
@@ -14,11 +14,23 @@ categories = ["asynchronous"]
 [dependencies]
 log = "0.4.22"
 
+# Requires `--cfg compute_heavy_executor_tokio` to enable.
 [target.'cfg(compute_heavy_executor_tokio)'.dependencies]
 tokio = { version = "1.0", features = ["rt", "rt-multi-thread", "macros", "sync"]}
 libc = { version = "0.2.168"}
 num_cpus = { version = "1.0"}
 
+[package.metadata.docs.rs]
+# enable tokio conditional compilation in the documentation
+rustdoc-args = ["--cfg", "compute_heavy_executor_tokio"]
+# it's necessary to _also_ pass `--cfg compute_heavy_executor_tokio`
+# to rustc, or else dependencies will not be enabled, and the docs build will fail.
+rustc-args = ["--cfg", "compute_heavy_executor_tokio"]
+
+[dev-dependencies]
+tokio = { version = "1.0", features = ["full"]}
+
 [lints.rust]
 # calling libraries can use the convention of `cfg(compute_heavy_executor)` to enable usage of this crate
-unexpected_cfgs = { level = "warn", check-cfg = ['cfg(compute_heavy_executor)', 'cfg(compute_heavy_executor_tokio)'] }
\ No newline at end of file
+unexpected_cfgs = { level = "warn", check-cfg = ['cfg(compute_heavy_executor)', 'cfg(compute_heavy_executor_tokio)'] }
```

Comment:
Per discussion, probably we should switch these to feature flags since anyway caller will want to directly depend on `compute-heavy-future-executor` to enable the strategy. And then they can add a direct dependency on eg `tokio-rustls` with the feature enabled to flip it over to using it.

Only issue would be if there were multiple MVs of this crate.